### PR TITLE
Bulk actions for users

### PR DESCRIPTION
**Pull request by KIRA009**
This PR adds bulk actions for the Users listing

image

### CODE REVIEW

File: wagtail/users/views/bulk_actions/toggle_activity.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    @classmethod
+    def execute_action(cls, objects, **kwargs):
+        cls.mark_as_active = kwargs.get('mark_as_active', False)
+        user = kwargs.get('user', None)
+        if user is not None:
+            objects = list(filter(lambda x: x.pk != user.pk, objects))
+        for user in objects:
+            user.is_active = cls.mark_as_active
+            user.save()
```
Comment (jacobtoppm) [inline wagtail/users/views/bulk_actions/toggle_activity.py:63]: We should consider using `bulk_update` for these - let's bring up in relation to logging at the core team meeting as we discussed

File: wagtail/users/views/bulk_actions/toggle_activity.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                    context['users'] = list(filter(lambda x: x['user'].pk != user.pk, context['users']))
+                    context['mark_self_as_inactive'] = [user]
+                    return TemplateResponse(self.request, self.template_name, context)
+
+    @classmethod
+    def execute_action(cls, objects, **kwargs):
+        cls.mark_as_active = kwargs.get('mark_as_active', False)
+        user = kwargs.get('user', None)
+        if user is not None:
+            objects = list(filter(lambda x: x.pk != user.pk, objects))
```
Comment (jacobtoppm) [inline wagtail/users/views/bulk_actions/toggle_activity.py:60]: I feel like permissions checks should already have taken place at this point: this is the part that could be run from a management command, so should just assume the user can perform the action

File: wagtail/users/views/bulk_actions/toggle_activity.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+from django.template.response import TemplateResponse
+from django.utils.translation import gettext_lazy as _
+from django.utils.translation import ngettext
+
+from wagtail.core import hooks
+from wagtail.core.compat import AUTH_USER_APP_LABEL, AUTH_USER_MODEL_NAME
+from wagtail.users.views.bulk_actions.user_bulk_action import UserBulkAction
+
+
+change_user_perm = "{0}.change_{1}".format(AUTH_USER_APP_LABEL, AUTH_USER_MODEL_NAME.lower())
```
Comment (jacobtoppm) [inline wagtail/users/views/bulk_actions/toggle_activity.py:11]: Why not import from https://github.com/wagtail/wagtail/blob/main/wagtail/users/views/users.py#L29?

File: wagtail/users/views/bulk_actions/assign_role.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+from django.template.response import TemplateResponse
+from django.utils.translation import gettext_lazy as _
+from django.utils.translation import ngettext
+
+from wagtail.core import hooks
+from wagtail.core.compat import AUTH_USER_APP_LABEL, AUTH_USER_MODEL_NAME
+from wagtail.users.views.bulk_actions.user_bulk_action import UserBulkAction
+
+
+change_user_perm = "{0}.change_{1}".format(AUTH_USER_APP_LABEL, AUTH_USER_MODEL_NAME.lower())
```
Comment (jacobtoppm) [inline wagtail/users/views/bulk_actions/assign_role.py:12]: You could import this from https://github.com/wagtail/wagtail/blob/main/wagtail/users/views/users.py#L29

**Review by jacobtoppm — CHANGES_REQUESTED** [review]
Generally looking great! A few things to clear up however

File: wagtail/users/views/bulk_actions/toggle_activity.py (hunk 2 above)
Comment (KIRA009) [reply wagtail/users/views/bulk_actions/toggle_activity.py:60]: It is checked in the `prepare_action` method, but the list of users that is passed on to `execute_action` is not modified. So, we have to remove the user from the list of users again

File: wagtail/users/views/bulk_actions/assign_role.py [hunk 5]
```diff
@@ -44,7 +44,7 @@ def prepare_action(self, objects):
 
     @classmethod
     def execute_action(cls, objects, **kwargs):
-        cls.role = role = kwargs.get('role', None)
+        role = kwargs.get('role', None)
```
Comment (jacobtoppm) [inline wagtail/users/views/bulk_actions/assign_role.py:47]: I feel like it's clearer to have these explicitly spelled out in the kwargs if they're a key part of the action

File: wagtail/users/views/bulk_actions/toggle_activity.py [hunk 6]
```diff
@@ -49,17 +49,17 @@ def prepare_action(self, objects):
...
             objects = list(filter(lambda x: x.pk != user.pk, objects))
         for user in objects:
-            user.is_active = cls.mark_as_active
+            user.is_active = mark_as_active
             user.save()
             cls.num_parent_objects += 1
 
     def get_success_message(self):
-        activity_status = "active" if self.mark_as_active else "inactive"
+        activity_status = "active" if self.cleaned_form.cleaned_data['mark_as_active'] else "inactive"
```
Comment (jacobtoppm) [inline wagtail/users/views/bulk_actions/toggle_activity.py:58]: This won't be translated, will it? I think you'll need to write `get_success_message` in a way where both the `activity_status` and the rest of the string will be translated, optimally together

**Review by jacobtoppm — CHANGES_REQUESTED** [review]
Very close now! A couple of minor bugfixes

**Review by jacobtoppm — APPROVED** [review]

**Comment by zerolab** [comment]
Thank you @KIRA009 and @jacobtoppm

**Pull request merged**